crates/data_components/Cargo.toml

```diff
@@ -43,6 +43,8 @@ arrow_sql_gen = { path = "../arrow_sql_gen", optional = true }
 rusqlite = { workspace = true, optional = true }
 tokio-rusqlite = { workspace = true, optional = true }
 spark-connect-rs =  { git = "https://github.com/edmondop/spark-connect-rs.git", rev = "24b8a437af41989bf7594e1606c37f30c134bba2", features = ["tls"], optional = true }
+odbc-api = { version = "7.0.0", optional = true }
+arrow-odbc = { version = "8.3.0", optional = true }
```

Comment:
Better to add this dependencies to workspace Cargo.toml, then
`arrow-odbc = { workspace = true, optional = true }`